Mathlib/CategoryTheory/ConnectedComponents.lean

```diff
@@ -40,6 +40,18 @@ def ConnectedComponents (J : Type u₁) [Category.{v₁} J] : Type u₁ :=
 instance [Inhabited J] : Inhabited (ConnectedComponents J) :=
   ⟨Quotient.mk'' default⟩
 
+/-- Every function from connected components of a category gives a functor to discrete category -/
+def ConnectedComponents.connectedToDiscrete.{u_1, u_2} {X C : Type u_1} [Category.{u_2, u_1} C]
+    (f : ConnectedComponents C -> X) : (C ⥤ Discrete X) where
+  obj :=  Discrete.mk ∘ f ∘ Quotient.mk (Zigzag.setoid _)
+  map :=  Discrete.eqToHom ∘ congrArg f ∘ Quotient.sound ∘ Zigzag.of_hom
+
+/-- Every functor to a discrete category gives a function from connected components -/
+def ConnectedComponents.discreteToConnected  {X C} [Category C] (fctr :C ⥤ Discrete X) :
```

Comment:
- Again, the name doesn't really make sense, something like `ConnectedComponents.discreteFunctorLift` would be better.
- `F` would be a better variable name than `fctr` for a functor. 
- The variable `C` and the category instance are already in scope, so you don't need to declare them again. In fact, `C` is never used throughout the file, so you might as well remove the variable line defining `C` and use `J` instead for the category, which is consistent with the rest of the file.

Reply:
so if the previous one was `functorToDiscrete`, as per the result type, this one should probably be `functionFromConnected`  ?

Reply:
Well, `functionFromConnected` could mean any function from the connected components. I think `discreteFunctorLift` (or something including `functor` and `lift`) is more descriptive because you're lifting a functor to a function on the connected components

Reply:
What about `toFunctorToDiscrete` and `fromFunctorToDiscrete` ?

Reply:
The following pairs would be consistent 

or `fromFunctiontoConnectedComponents` and `fromFunctorToDiscrete`

or `toFunctorToDiscrete` and `fromFunctorToDiscrete`

or `fromFunctiontoConnectedComponents  and `toFunctiontoConnectedComponents`

or `toFunctorToDiscrete ` and `toFunctiontoConnectedComponents `